api/handlers.go

```diff
@@ -742,6 +743,14 @@ func (api *API) userPUT(u *database.User, w http.ResponseWriter, req *http.Reque
 		changedEmail = true
 	}
 
+	if api.staticDeps.Disrupt("DependencyUserPutMongoDelay") {
+		time.Sleep(100 * time.Millisecond)
+	}
+	if api.staticDeps.Disrupt("DependencyMongoWriteConflictN") {
```

Comment:
This is a bit funky? This is essentially a disrupt that returns a custom error message, and then you have a unit test that sets the dependency, call the route and verify you get a failure N times, and then a success? That's really only a test to make sure this `Disrupt` call works, and I guess aside from that it's also asserting the code works if you ensure there's only one transaction to the user object.

Reply:
I expected to see the dependency closer to the database, ideally before an update or right after it, so it kicks in after `N` times and returns the error after an interaction with a database, rather than at the top of a handler.

Reply:
Done.